src/components/grid-smart/grid-smart.tsx

```diff
@@ -359,15 +367,31 @@ export class GridSmart
     this.swiper.allowTouchMove = !lock;
   }
 
-  private initSwiper() {
-    if (this.swiper == null && this.recordCount > 0) {
+  private ensureSwiper(): boolean {
+    if (
+      !this.swiper &&
+      this.recordCount > 0 &&
+      this.loadingState !== "loading"
+    ) {
+      const opts: SwiperOptions = this.normalizeOptions();
       const container: HTMLElement = this.el;
       container
         .querySelector("[slot='grid-content']")
         .classList.add("swiper-wrapper");
-      this.swiper = new Swiper(container, this.normalizeOptions());
+      this.log("Initializing Swiper..");
+      this.swiper = new Swiper(container, opts);
+    }
+    return this.isInitialized();
+  }
+
+  private isInitialized(): boolean {
+    return this.swiper !== null;
+  }
+
+  private log(msg: any): void {
+    if (msg && this.logLevel !== "off") {
+      // console.log(msg, this.recordCount, this.loadingState, this.el);
```

Comment:
I'm not sure about leaving this commented code here.

Reply:
This will be resolved later. After introducing logging api.